src/Kernel.php

```diff
@@ -145,9 +148,15 @@ function pipleline(callable $firstCallback, callable ...$callbacks): callable
 
 const loadState = 'Prooph\Micro\Kernel\loadState';
 
-function loadState(Message $message, AggregateDefiniton $definiton): array
+function loadState(SnapshotStore $snapshotStore, Message $message, AggregateDefiniton $definiton): array
 {
-    return []; // @todo: fetch from projections
+    $aggregate = $snapshotStore->get($definiton->aggregateType(), $definiton->extractAggregateId($message));
+
+    if (! $aggregate) {
+        return [];
+    }
+
+    return $aggregate->aggregateRoot();
```

Comment:
The snapshot-store is also used in the event-sourcing lib. I could not come up with a name that fits both perfectly.